### PR TITLE
Week03 submissions

### PR DESCRIPTION
**Pull request by m-akash**
Hi @Sadman007, please review my submissions for week 03.  thanks!

### CODE REVIEW

File: Week03/Implement Trie (Prefix Tree).cpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
+/*
+Complexity Analysis
+Time Complexity:
+Insert/Search/StartsWith: O(M) per operation.
+Overral: O(M * N).
+
+Space Complexity: O(M * N).
+*/
```
Comment (Sadman007) [inline Week03/Implement Trie (Prefix Tree).cpp:8]: What is M and N?

File: Week03/Implement Trie (Prefix Tree).cpp [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+class Trie {
+public:
+    class TrieNode {
+    public:
+        char ch;
```
Comment (Sadman007) [inline Week03/Implement Trie (Prefix Tree).cpp:13]: Unused. Remove it.

File: Week03/Maximum XOR of Two Numbers in an Array.cpp [hunk 3]
```diff
@@ -0,0 +1,75 @@
+/*
+Complexity Analysis
+Time Complexity:
+Trie Insertion: O(N * B), where N is the size of nums, and B = 31 is the bit length.
+Best Match Queries: O(N * B)
```
Comment (Sadman007) [inline Week03/Maximum XOR of Two Numbers in an Array.cpp:5]: Both are same. So it's redundant to define `Best Match Queries` separately. Remove it.

**Review by Sadman007 — APPROVED** [review]
Score: 10/10

Nice work!

File: Week03/Implement Trie (Prefix Tree).cpp (hunk 1 above)
Comment (m-akash) [reply Week03/Implement Trie (Prefix Tree).cpp:8]: Here, M Represents the average length of the words being inserted in the Trie, and N  Represents the total number of words processed by the Trie.

**Pull request merged**